vivisect/analysis/generic/codeblocks.py

```diff
@@ -129,16 +131,21 @@ def analyzeFunction(vw, funcva):
 
         # So we don't add a codeblock if we're re-analyzing a function
         # (like during dynamic branch analysis)
-        bsize = blocks[bva]
-        tmpcb = vw.getCodeBlock(bva)
-        if bva not in oldblocks or tmpcb is None:   # sometimes codeblocks can be deleted if owned by multiple functions
-            vw.addCodeBlock(bva, bsize, funcva)
-        elif bsize != oldblocks[bva]:
-            vw.delCodeBlock(bva)
-            vw.addCodeBlock(bva, bsize, funcva)
-        bcnt += 1
+        try:
+            bsize = blocks[bva]
+            tmpcb = vw.getCodeBlock(bva)
+            # sometimes codeblocks can be deleted if owned by multiple functions
+            if bva not in oldblocks or tmpcb is None:
+                vw.addCodeBlock(bva, bsize, funcva)
+            elif bsize != oldblocks[bva]:
+                vw.delCodeBlock(bva)
+                vw.addCodeBlock(bva, bsize, funcva)
+            bcnt += 1
+        except Exception as e:
+            logger.warning('Codeblock analysis hit exception: %s', e)
+            break
 
     vw.setFunctionMeta(funcva, 'Size', size)
     vw.setFunctionMeta(funcva, 'BlockCount', bcnt)
     vw.setFunctionMeta(funcva, 'InstructionCount', opcount)
-    vw.setFunctionMeta(funcva, 'MnemDist', mnem)
+    vw.setFunctionMeta(funcva, 'MnemDist', dict(mnem))
```

Comment:
is this a "best practices" thing?  why would we create a new dict, when the old dict is not going to be used again?

Reply:
It's actually a minor display thing. I got tired of seeing the `collections.defaultdict(<int>, {......}` stuff in the output UI since it disables the pretty printing of that.